### PR TITLE
Add a roadmap to OpenShift

### PR DESCRIPTION
**Pull request by smarterclayton**
Covers high level objectives and important cross project initiatives.

/assign @derekwaynecarr

I don't think we have to complete this before merging it, it would be better to have TODOs and add those slowly. However getting the use cases and objectives down is key.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+This lists the important initiatives across the project. These are the ones that span
+multiple releases, require close coordination between teams, or have subtle implications
+on a large number of areas.
+
+
+### Automating management of the control plane
+
+Our goal is to fully automate control plane node lifecycle, reduce operational complexity
+during recovery of a master, simplify the install sequence and remove the need for a
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply ROADMAP.md]: In the long run it is very desirable.  How about we qualify that this is a multi release timeline, with the goal to continue to simplify the requirements for our install process (which includes the example of removing bootstrap)?

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+I should expect Kubernetes and the necessary ecosystem components to remain stable over multiple-year timeframes,
+so that I can delivery applications more rapidly, with better operational efficiency, at higher scales, and with better availability.
+
+This includes:
+
+* API stability and conformance within the Kubernetes project and other ecosystem projects
+* Backwards and forwards compatibility for all APIs and extensions - all breaks are regressions
+* A clear lifecycle that matches my organizational needs with safe upgrades and long term support
+* Automation for common operational patterns like autoscaling, machine lifecycle, and load balancer integration
+* Automatic hardware, infrastructure, and softweare monitoring and remediation to mitigate entropy
```
Comment (enxebre) [inline ROADMAP.md]: typo `softweare`

File: ROADMAP.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply ROADMAP.md]: i'm working on adding a doc that provides that context why keeping bootstrap host is useful today.. and maybe then we can talk about what we are ready to give up to get to no bootstrap

[the remark this answers is not in the document]
Comment (smarterclayton) [reply ROADMAP.md]: Crawford and I talked, I'm going to change this to "make it easy to convert the bootstrap node into a worker or secondary master".  We'll create a different path for "single node" clusters that will be called out in another section.

**Comment by smarterclayton** [comment]
Going to merge this with the minor edits suggested, we can continue to iterate on this but I want to have something concrete checked in.

**Pull request merged**